src/atlas_schema/utils.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import annotations
+
+from typing import TypeVar, Union, cast
+
+import awkward as ak
+import dask_awkward as dak
+
+Array = TypeVar("Array", bound=Union[dak.Array, ak.Array])
+
+
+def isin(haystack: Array, needles: dak.Array | ak.Array) -> Array:
+    """
+    Find needles in haystack.
+
+    This works by first transforming haystack to Array[N*M*K] and then doing a comparison.
+
+    Args:
+        haystack (dak.Array[N*M] or ak.Array[N*M]): three-dimensional haystack of values.
```

Comment:
Is this limited to `N*M` shapes and does not generalize higher than that?

Reply:
I also would call this two-dimensional (where the value at each slot is not an extra "dimension").

Reply:
Right now, I think so. I am going to see if I can make it work to broadcast through to the last dimension of the haystack -- meaning I need to look up `ndim` and see if I can dynamically expand that.

Reply:
yes, I can generalize it.